cli/src/potpie/server_manager.py

```
@@ -0,0 +1,373 @@
+            if attempt >= max_attempts:
+                logging.error(
+                    "Docker containers does not run within the time 60 seconds"
+                )
+
+                raise DockerError("Docker containers failed to start in time.")
+
+            logging.info("Docker containers started successfully")
+
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Failed to start Docker containers: {e}")
+            raise DockerError("Failed to start Docker containers")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)